bin/schema_generator.sh

```diff
@@ -56,124 +52,72 @@ function setup_git_ssh() {
+    git checkout "$mps_branch_base"
+    git branch -D "$mps_branch_temp" || :
+    git checkout -b "$mps_branch_temp"
+    popd
+
+    validate_bigquery copy
+    # Generate schemas, commits results to the publish branch
+    generate_commit "$mps_root" "$mps_branch_source" "$mps_branch_publish"
+    validate_bigquery copy
+    # Run the test.
+    validate_bigquery local --head "$mps_branch_publish" --base "$mps_branch_temp"
 }
 
 function main() {
-    cd $BASE_DIR
-
-    # Setup ssh key and git config
-    setup_git_ssh
-
-    # Pull in all schemas from MPS and change directory
-    clone_and_configure_mps
-
-    # Generate concrete JSON schemas that contain per-probe fields.
-    # These are used only as the basis for generating BQ schemas;
-    # we publish JSON schemas exactly as they appear in the source branch so
-    # that the pipeline doesn't rely on per-probe types when validating pings.
-    # For Glean pings, we copy the generic Glean schema into place later on.
-    mozilla-schema-generator generate-main-ping --out-dir ./telemetry --mps-branch $MPS_BRANCH_SOURCE
-    mozilla-schema-generator generate-common-pings --common-pings-config $COMMON_PINGS_PATH --mps-branch $MPS_BRANCH_SOURCE --out-dir ./telemetry
-    mozilla-schema-generator generate-glean-pings --mps-branch $MPS_BRANCH_SOURCE --out-dir .
-
-    # Remove all non-json schemas (e.g. parquet)
-    find . -not -name "*.schema.json" -type f -exec rm {} +
-
-    # Add metadata fields to all json schemas
-    # schema directory structure is enforced by regex at compile-time
-    # shellcheck disable=SC2044
-    for schema in $(find . -name "*.schema.json" -type f); do
-        metadata_merge metadata/ "$schema"
-    done
-
-    # Add transpiled BQ schemas
-    find . -type f -name "*.schema.json" | while read -r fname; do
-        # This schema is AWS-specific, fails transpilation, and should be ignored
-        if [[ $fname =~ metadata/sources ]] ; then
-            continue
-        fi
-        bq_out=${fname/schema.json/bq}
-        mkdir -p $(dirname "$bq_out")
-        jsonschema-transpiler \
-            --resolve drop \
-            --type bigquery \
-            --normalize-case \
-            --force-nullable \
-            --tuple-struct \
-                "$fname" > "$bq_out"
-    done
-
-    # Keep only allowed schemas
-    filter_schemas
-
-    # Copy aliased BQ schemas into place
-    alias_schemas $ALIASES_PATH .
-
-    # Push to branch of MPS
-    cd ../
-    commit_schemas
+    pushd .
+    # the base directory in the docker container
+    cd /app
+    if ! stat "$HOME/.ssh/id_ed25519"; then
+        setup_git
+    fi
+    setup_mps
+
+    validate_generate_commit \
+        /app/mozilla-pipeline-schemas \
+        "$MPS_BRANCH_SOURCE" \
+        "$MPS_BRANCH_PUBLISH"
+
+    cd mozilla-pipeline-schemas
     git push || git push --set-upstream origin "$MPS_BRANCH_PUBLISH"
+    popd
 }
 
-main "$@"
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
```

Comment:
This is a big improvement for testing, good call. It was always annoying that I had to copy+paste the functions into the container